app/controllers/api/v1/shoutouts_controller.rb

```diff
@@ -0,0 +1,51 @@
+require 'digest'
+
+class Api::V1::ShoutoutsController < ApplicationController
+  include ApplicationHelper
+  before_action :require_user!
+
+  def create
+    @shoutout = Shoutout.new(shoutout_params)
```

Comment:
I tried to change at the ```
 def create
    @shoutout = current_user.shoutouts.new(shoutout_params)
    if @shoutout.save
```
but I got mistake in the test ```
  1) Api::V1::ShoutoutsController POST #create with invalid parameters does not create a shout-out with invalid user_id
     Failure/Error:
       expect do
         post :create, params: { shoutout: { user_id: nil,
                                             time_period_id: time_period.id,
                                             rich_text: 'value2',
                                             recipients: [user2.id, user3.id, user4.id] } }
       end.not_to change(Shoutout, :count)
     
       expected `Shoutout.count` not to have changed, but did change from 0 to 1
```
What is wrong in my act the test or changing way of got @shoutout?

Reply:
@VitaliiShevchenko try removing the `user_id` from the parameters being passed. Note that you'll also need to prevent it from being sent from the front-end.